### ISSUE TITLE
Host key verification failed

### ISSUE THREAD

**Issue by blindmonkey** [peripheryapp/periphery#661]
We've been using Periphery in our project for a while now, both in Xcode and on CI. Although we've encountered some issues previously (including this one), we've thus far been able to resolve them. But this time I'm stumped.

The error that we get when running Periphery is hard to identify, but it seems logical that it would be the same error that we'd get when just running the command in isolation.
```
error: Shell command '/usr/bin/env xcodebuild -project [REDACTED].xcodeproj -showBuildSettings -json -target [REDACTED] -target [REDACTED]Tests -target [REDACTED]HostedTests -target [REDACTED]Demo' returned exit status '74':
```

Running the command on its own also fails and this time actually prints the underlying error:
```
$ xcodebuild -workspace [REDACTED].xcworkspace -list -json
xcodebuild: error: Could not resolve package dependencies:
  Failed to clone repository git@github.com:[REDACTED]/[REDACTED].git:
    Cloning into bare repository '[REDACTED]'...
    Host key verification failed.
fatal: Could not read from remote repository.
    Please make sure you have the correct access rights
    and the repository exists.
```

As I mentioned, we've previously gotten this error, though it was for the `/usr/bin/env xcodebuild -workspace [REDACTED] -list -json` command. We managed to work around it by running the command directly before running Periphery, but this recently stopped working.

The repository in question is an SPM dependency hosted in a private repo, for which we use an SSH key added to `ssh-agent`. Our build commands specify both a `-derivedDataPath` (to ensure repeatable clean builds) and `-clonedSourcePackagesDirPath` (to cache our SPM dependencies). To ensure that things run properly on CI we also specify `-scmProvider system -usePackageSupportBuiltinSCM` arguments, and just to be completely sure we also run `defaults write com.apple.dt.Xcode IDEPackageSupportUseBuiltinSCM YES`.

None of these options seem to be specifiable to Periphery, and the options in `com.apple.dt.Xcode` (both in `~/Library/Preferences` and `/Library/Preferences`) don't appear to be getting picked up.

This issue happens consistently both on the latest release (2.16.0) and on the version we were previously using (2.10.2).

**Comment by blindmonkey** [comment]
I was able to work around the initial issue I reported by adding the key via `/usr/bin/env ssh-add`. It's not ideal to have to do this on CI, but it was a step forward. However, now Periphery is refusing to read the DataStore. It's not really clear why this would be, since not much has changed about our pipeline with the exception of provisioning a new machine for CI and that machine being on macOS Sonoma. We're using Xcode 14.3 with the intention of updating to 15.0 shortly.

When running in verbose mode, I'm seeing a number of errors like `[index:swift] Source file not indexed: /Users/[REDACTED].swift`, as well as a general error:
```
error: The index store at '["/Users/[REDACTED]/derivedData/Index.noindex/DataStore"]' does not contain data for the following targets: [REDACTED TARGETS]. Either the index store is outdated, or you have requested to scan targets that have not been built. For Xcode projects, the chosen schemes must build all of the chosen targets.
```

I spot checked the index store and it seems to have data (though I don't know how to read it). I was thinking that maybe it's a permissions issue, but I would've expected to see a more specific error around that if that were the case.

Any help or advice would be much appreciated.

Thank you in advance!

**Comment by ileitch** [comment]
> Edit: Looks like the original issue wasn't fully resolved after all. I'm still getting errors about fetching our private SPM dependencies, it just doesn't always happen.

Are these errors happening when running `xodebuild` directly? If so, I think that needs to be resolved first before trying to diagnose the Periphery failures.

**Comment by blindmonkey** [comment]
> > Edit: Looks like the original issue wasn't fully resolved after all. I'm still getting errors about fetching our private SPM dependencies, it just doesn't always happen.
> 
> Are these errors happening when running `xodebuild` directly? If so, I think that needs to be resolved first before trying to diagnose the Periphery failures.

No, I'm not getting these errors directly. I was for a bit, but that was due to a misconfiguration with our build machines. Once I fixed that, I'm now able to run the `xcodebuild` commands directly but Periphery fails. My current workaround is to run `$ xcodebuild -workspace [REDACTED].xcworkspace -list -json` before running Periphery.

**Comment by ileitch** [comment]
This issue should now be resolved in 3.0.0. Let me know if you continue to experience this issue and I will reinvestigate.
